### PR TITLE
apps: look for existing brick remove ops and fail gracefully if so

### PR DESCRIPTION
**Pull request by phlogistonjohn**
The existing check for pending operations on bricks was missing
any non-brick operations on the device, specifically other
existing pending device remove operations. Add a check and a
test for that condition.

Signed-off-by: John Mulligan <jmulligan@redhat.com>

### CODE REVIEW

**Comment by obnoxxx** [comment]
```
# github.com/heketi/heketi/apps/glusterfs
apps/glusterfs/operations_test.go:1656:17: too many arguments in call to v.Create
apps/glusterfs/operations_test.go:1656:43: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1681:18: too many arguments in call to d.SetState
apps/glusterfs/operations_test.go:1681:44: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1685:33: too many arguments in call to NewDeviceRemoveOperation
apps/glusterfs/operations_test.go:1685:48: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1686:17: too many arguments in call to dro.Build
apps/glusterfs/operations_test.go:1686:21: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1693:34: too many arguments in call to NewDeviceRemoveOperation
apps/glusterfs/operations_test.go:1693:49: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1693:34: too many errors
failed ./tests/030-govet.sh [2]
```

**Comment by obnoxxx** [comment]
```
-- Running: 100-gotest.sh --
--- testing: github.com/heketi/heketi ---
?   	github.com/heketi/heketi	[no test files]
total:	(statements)	0.0%
--- testing: github.com/heketi/heketi/apps ---
?   	github.com/heketi/heketi/apps	[no test files]
total:	(statements)	0.0%
--- testing: github.com/heketi/heketi/apps/glusterfs ---
# github.com/heketi/heketi/apps/glusterfs
apps/glusterfs/operations_test.go:1656:17: too many arguments in call to v.Create
apps/glusterfs/operations_test.go:1656:43: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1681:18: too many arguments in call to d.SetState
apps/glusterfs/operations_test.go:1681:44: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1685:33: too many arguments in call to NewDeviceRemoveOperation
apps/glusterfs/operations_test.go:1685:48: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1686:17: too many arguments in call to dro.Build
apps/glusterfs/operations_test.go:1686:21: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1693:34: too many arguments in call to NewDeviceRemoveOperation
apps/glusterfs/operations_test.go:1693:49: app.Allocator undefined (type *App has no field or method Allocator)
apps/glusterfs/operations_test.go:1693:34: too many errors
FAIL	github.com/heketi/heketi/apps/glusterfs [build failed]
```

**Comment by phlogistonjohn** [comment]
Oh, dear. I moved the patch to the wrong place. Will fix.

File: apps/glusterfs/operations_test.go [hunk 1]
```diff
@@ -1628,6 +1628,83 @@ func TestDeviceRemoveOperationOtherPendingOps(t *testing.T) {
...
+	err = dro.Build()
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	// perform the build step of a 2nd remove operation
+	// we can "fake' it this way in a test because the transactions
+	// that cover the Build steps are effectively serializing
+	// these actions.
+	dro2 := NewDeviceRemoveOperation(d.Info.Id, app.db)
+	err = dro2.Build()
+	tests.Assert(t, err != nil, "expected err != nil, got:", err)
```
Comment (obnoxxx) [inline apps/glusterfs/operations_test.go]: This first check seems redundant.

**Comment by phlogistonjohn** [comment]
Redundant check removed. I looked fo a typo but did't see it. I'm sure it will be obvious once you point it out, though.

File: apps/glusterfs/operations_test.go [hunk 2]
```diff
@@ -1628,6 +1628,82 @@ func TestDeviceRemoveOperationOtherPendingOps(t *testing.T) {
...
+	vreq.Durability.Type = api.DurabilityReplicate
+	vreq.Durability.Replicate.Replica = 3
+	// create volumes
+	for i := 0; i < 4; i++ {
+		v := NewVolumeEntryFromRequest(vreq)
+		err = v.Create(app.db, app.executor)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	}
+
+	// grab a devices that has bricks
```
Comment (obnoxxx) [inline apps/glusterfs/operations_test.go]: typo: devices --> device

**Pull request merged**